pytorch_lightning/core/lightning.py

```diff
@@ -92,19 +92,19 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self._device_type = None
 
         #: True if using amp
-        self.use_amp = False
+        self.use_amp: bool = False
 
         #: The precision used
-        self.precision = 32
+        self.precision: int = 32
 
         # optionally can be set by user
         self._example_input_array = None
         self._datamodule = None
         self._results: Optional[Result] = None
-        self._current_fx_name = ''
-        self._running_manual_backward = False
-        self._current_hook_fx_name = None
-        self._current_dataloader_idx = None
+        self._current_fx_name: str = ''
+        self._running_manual_backward: bool = False
+        self._current_hook_fx_name: Optional[str] = None
+        self._current_dataloader_idx: Optional[int] = None
```

Comment:
Style question: Do you think it's worth annotating variables with clear types?

```python
x = 1  # type will always be int, not worth annotating
x: Optional[int] = 1  # As its default is a different type, there's value for annotating
```

Reply:
i think it's always helpful to annotate just in case another part of the code holds doesn't keep the same assumption. checking none vs empty str here is one example